buildSrc/src/main/kotlin/otel.publish-conventions.gradle.kts

```diff
@@ -63,3 +80,36 @@ if (System.getenv("CI") != null) {
     sign(publishing.publications["mavenPublication"])
   }
 }
+
+if (!project.name.startsWith("bom")) {
+  project.afterEvaluate {
+    val sbomName = "opentelemetry-java_" + base.archivesName.get()
+    tasks.cyclonedxBom {
+      outputName = "$sbomName.cyclonedx"
+      includeConfigs = listOf("runtimeClasspath")
+      skipConfigs = listOf("compileClasspath", "testCompileClasspath")
+      destination = project.file("build/cyclonedx")
+      outputFormat = "json"
+    }
+    spdxSbom {
+      targets {
+        // Create a target to match the published jar name.
+        // This is used for the task name (spdxSbomFor<SbomName>)
+        // and output file (<sbomName>.spdx.json).
+        create(sbomName) {
+          scm {
+            uri.set("https://github.com/" + System.getenv("GITHUB_REPOSITORY"))
+            revision.set(System.getenv("GITHUB_SHA"))
+          }
+          document {
+            name.set(sbomName)
+            namespace.set("https://opentelemetry.io/spdx/" + UUID.randomUUID())
```

Comment:
What's the significance of this UUID? Does this URL need to resolve? Does the UUID need to be stable across releases?

Reply:
No, I've just been told it is supposed to be unique.

For reference:
https://spdx.github.io/spdx-spec/v2.3/document-creation-information/#651-description

Reply:
Per the documentation, the intent is to "provide an unambiguous mechanism for other SPDX documents to reference SPDX elements within this SPDX document". 
and
"The URI does not have to be accessible. It is only intended to provide a unique ID"
https://spdx.github.io/spdx-spec/v2.3/document-creation-information/#652-intent